bootstrap: be more and less verbose during script execution

NicolasT opened the pull request:
Successful execution (on, admittedly, a previously-deployed bootstrap node, hence the timings):

```
$ ./bootstrap.sh
> Pre-minion system tests... done [0s]
> Disabling Salt minion service... done [0s]
> Stopping Salt minion service... done [1s]
> Configuring local YUM repositories... done [0s]
> Installing Salt minion... done [0s]
> Configuring Salt minion to run in local mode... done [4s]
> Calculating Salt grains in local mode... done [2s]
> Deploying early-stage bootstrap node in local mode (this may take a while)... done [14s]
> Syncing Pillar modules on Salt master... done [1s]
> Syncing Runner modules on Salt master... done [2s]
> Accepting bootstrap minion key on Salt master... done [3s]
> Deploying bootstrap node (this may take a while)... done [48s]
```

When failing:

```
$ SALT_CALL=not-salt-call ./bootstrap.sh
> Pre-minion system tests... done [0s]
> Disabling Salt minion service... done [0s]
> Stopping Salt minion service... done [0s]
> Configuring local YUM repositories... done [1s]
> Installing Salt minion... done [0s]
> Configuring Salt minion to run in local mode... done [0s]
> Calculating Salt grains in local mode... fail [0s]

Failure while running step 'Calculating Salt grains in local mode'

Command: not-salt-call --local --state-output=mixed --retcode-passthrough state.sls metalk8s.node.grains saltenv=metalk8s-2.0 pillarenv=metalk8s-2.0

Output:

<< BEGIN >>
./bootstrap.sh: line 48: not-salt-call: command not found
<< END >>

This script will now exit

```

Fixes: #1029
Fixes: https://github.com/scality/metalk8s/issues/1029

NicolasT committed:
bootstrap: be more and less verbose during script execution

Fixes: #1029
Fixes: https://github.com/scality/metalk8s/issues/1029

bert-e commented:
# Hello nicolast,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1029-bootstrap-verbosity` into
`development/2.0`
* [w/2.1/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1029-bootstrap-verbosity) will be merged into `development/2.1` 
* [w/2.2/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1029-bootstrap-verbosity) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

gdemonet commented on scripts/bootstrap.sh.in:
```diff
@@ -29,6 +37,50 @@ declare -A GPGCHECK_REPOSITORIES=(
+    local start
+    start=$(date +%s)
+    set +e
+    "$@" > "${TMPFILES}/out" 2>&1
```
Could we toggle this redirection based on environment variable/CLI arg? Might be useful to follow what the bootstrap is doing, especially if we are running Salt with `-l debug`, or if we improve long-running commands to output progress information.

NicolasT replied:
Makes sense, will look into that.

gdemonet reviewed (dismissed):
LGTM - a minor comment about having the option to keep the output in stdout/stderr during a `run`, which may not be needed in this PR

TeddyAndrieux commented:
Could be cool to have the output in a log file something like `/var/log/metalk8s/bootstrap.log`

TeddyAndrieux commented:
Also not directly related to this PR but we should set the `log_level_logfile` of the salt-minion to at least `info` IMO

NicolasT committed:
salt: set Salt minion log-level to `info`

NicolasT committed:
bootstrap: add `--verbose` option

See: https://github.com/scality/metalk8s/pull/1032#pullrequestreview-230617371

bert-e commented:
# History mismatch



Merge commit #d5fec702708418f5141bdee41d9b9487006bff11 on the integration branch
`w/2.1/improvement/1029-bootstrap-verbosity` is merging a branch which is neither the current
branch `improvement/1029-bootstrap-verbosity` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1029-bootstrap-verbosity` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

NicolasT commented:
> LGTM - a minor comment about having the option to keep the output in stdout/stderr during a `run`, which may not be needed in this PR

Done

NicolasT commented:
> Also not directly related to this PR but we should set the `log_level_logfile` of the salt-minion to at least `info` IMO

Done

NicolasT commented:
> Could be cool to have the output in a log file something like `/var/log/metalk8s/bootstrap.log`

How would you code that up in Bash? `exec > >$(tee -ia /var/log/metalk8s/bootstrap.log)` or similar?

NicolasT commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1029-bootstrap-verbosity` into
`development/2.0`
* [w/2.1/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1029-bootstrap-verbosity) will be merged into `development/2.1` 
* [w/2.2/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1029-bootstrap-verbosity) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

bert-e commented:
# History mismatch



Merge commit #0d54094f45d845ade46daff96231de2962b9f8b2 on the integration branch
`w/2.1/improvement/1029-bootstrap-verbosity` is merging a branch which is neither the current
branch `improvement/1029-bootstrap-verbosity` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1029-bootstrap-verbosity` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

NicolasT commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1029-bootstrap-verbosity` into
`development/2.0`
* [w/2.1/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1029-bootstrap-verbosity) will be merged into `development/2.1` 
* [w/2.2/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1029-bootstrap-verbosity) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

TeddyAndrieux commented:
> > Could be cool to have the output in a log file something like `/var/log/metalk8s/bootstrap.log`
> 
> How would you code that up in Bash? `exec > >$(tee -ia /var/log/metalk8s/bootstrap.log)` or similar?

Yes indeed in bash it's ugly but we can do something like only have "less verbose" output in stdout and keep the verbose version in the log file

NicolasT committed:
bootstrap: log output by default

The default log-file location (`/var/log/metalk8s-bootstrap.log`) can be
overridden using the `--log-file` argument, e.g. sending it to
`/dev/null`.

See: https://github.com/scality/metalk8s/pull/1032#issuecomment-486665642

NicolasT commented:
> > > Could be cool to have the output in a log file something like `/var/log/metalk8s/bootstrap.log`
> > 
> > 
> > How would you code that up in Bash? `exec > >$(tee -ia /var/log/metalk8s/bootstrap.log)` or similar?
> 
> Yes indeed in bash it's ugly but we can do something like only have "less verbose" output in stdout and keep the verbose version in the log file

Done

NicolasT commented:
~Oops, forgot to redirect command output to log-file in silent mode.~ Done

bert-e commented:
# History mismatch



Merge commit #0e088c71408ab6196a9c4b135460866f42804dcf on the integration branch
`w/2.1/improvement/1029-bootstrap-verbosity` is merging a branch which is neither the current
branch `improvement/1029-bootstrap-verbosity` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1029-bootstrap-verbosity` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

NicolasT commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1029-bootstrap-verbosity` into
`development/2.0`
* [w/2.1/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1029-bootstrap-verbosity) will be merged into `development/2.1` 
* [w/2.2/improvement/1029-bootstrap-verbosity](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1029-bootstrap-verbosity) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

NicolasT commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **approve**

gdemonet reviewed (approved):
LGTM, tested and the output looks nice, the logfiles are well populated.

Really minor, but we have the command outputs starting on the same line as the `> ${command_name}...` in the `metalk8s-bootstrap.log` file (we may want to insert a newline for clarity, maybe later).

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye nicolast.
